### PR TITLE
@/roboshield Current footer to Payload 

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

- Introduces Richtext Component.
- Uses Site settings to populate Footer.
- Could have been completed with #733  but I had to branch off so it doesn't block other implementations.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
👍🏽 
---

I think lets just add one tab: **Initiative** (or **Project** if you think it makes more sense).
```js
  title: "", // e.g. In partnership with
  description: "", // e.g. This initiative ...
  partners: [], // list of partners
```

My main reason/thinking being these are the initiative/project partners so it make sense to have everything in one Tab. As a matter of fact, may be **Initiative**/**Project** should be its own setting outside of main **Site** settings...

**Comment by kilemensi** [comment]
pnpm dev is complaining about missing ts-node @koechkevin 
```sh
clemence@CFA-CLEMENCE roboshield % pnpm dev

> roboshield@0.1.0 dev /Users/clemence/Work/github.com/CodeForAfrica/ui/apps/roboshield
> NODE_OPTIONS='--inspect' ts-node --project tsconfig.server.json server.ts

sh: ts-node: command not found
 ELIFECYCLE  Command failed.
clemence@CFA-CLEMENCE roboshield %
```

This is after running `pnpm install --frozen-lockfile`.

**Comment by koechkevin** [comment]
> `pnpm dev` is complaining about missing `ts-node` @koechkevin
> 
> ```shell
> clemence@CFA-CLEMENCE roboshield % pnpm dev
> 
> > roboshield@0.1.0 dev /Users/clemence/Work/github.com/CodeForAfrica/ui/apps/roboshield
> > NODE_OPTIONS='--inspect' ts-node --project tsconfig.server.json server.ts
> 
> sh: ts-node: command not found
>  ELIFECYCLE  Command failed.
> clemence@CFA-CLEMENCE roboshield %
> ```
> 
> This is after running `pnpm install --frozen-lockfile`.

🤔

**Comment by koechkevin** [comment]
> > `pnpm dev` is complaining about missing `ts-node` @koechkevin
> > ```shell
> > clemence@CFA-CLEMENCE roboshield % pnpm dev
> > 
> > > roboshield@0.1.0 dev /Users/clemence/Work/github.com/CodeForAfrica/ui/apps/roboshield
> > > NODE_OPTIONS='--inspect' ts-node --project tsconfig.server.json server.ts
> > 
> > sh: ts-node: command not found
> >  ELIFECYCLE  Command failed.
> > clemence@CFA-CLEMENCE roboshield %
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This is after running `pnpm install --frozen-lockfile`.
> 
> 🤔

@kilemensi ts-node was missing in list of dev dependencies. I have added it.

File: apps/roboshield/src/lib/data/payload.types.ts [hunk 1]
```diff
@@ -74,6 +79,8 @@ export interface Settings {
     title: string;
     embedCode: string;
   };
+  partners: Partner[];
+  partnerHeaderTitle: string;
```
Comment (kilemensi) [inline apps/roboshield/src/lib/data/payload.types.ts:83]: Are these still needed?

File: apps/roboshield/src/components/Footer/FooterDescription.tsx [hunk 2]
```diff
@@ -32,18 +33,25 @@ function FooterDescription({ description, logo, sx }: FooterDescriptionProps) {
...
-      >
-        {description}
-      </RichTypography>
+        sx={(theme: Theme) => ({
+          a: {
+            color: theme.palette.text.secondary,
+            textDecorationColor: theme.palette.text.secondary,
+          },
+          mt: "52px",
+          textAlign: { xs: "center", md: "left" },
```
Comment (kilemensi) [inline apps/roboshield/src/components/Footer/FooterDescription.tsx:50]:
```suggestion
```

Even in mobile, paragraph text looks better when left aligned.


S

File: apps/roboshield/src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -116,20 +124,30 @@ export default function Footer({
...
-                >
-                  {project}
-                </RichTypography>
+                  sx={(theme: Theme) => ({
+                    a: {
+                      color: theme.palette.text.secondary,
+                      textDecorationColor: theme.palette.text.secondary,
+                    },
+                    mt: "52px",
+                    textAlign: { xs: "center", md: "left" },
```
Comment (kilemensi) [inline apps/roboshield/src/components/Footer/Footer.tsx:146]:
```suggestion
```

**Review by kilemensi — APPROVED** [review]

**Pull request merged**